sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Diagnostics/OptimizationsBaseEventSource.cs

```diff
@@ -0,0 +1,960 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Tracing;
+using System.Runtime.CompilerServices;
+using Azure.Core.Diagnostics;
+
+namespace Azure.Messaging.EventHubs.Diagnostics
+{
+    /*
+    See: https://learn.microsoft.com/en-us/dotnet/core/diagnostics/eventsource-instrumentation#optimizing-performance-for-high-volume-events
+    and https://learn.microsoft.com/en-us/dotnet/core/diagnostics/eventsource-instrumentation#eventsource-class-hierarchies
+
+    EventSource methods: https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventsource?view=net-9.0
+    */
+
+    internal abstract class OptimizationsBaseEventSource : AzureEventSource
```

Comment:
nitpicky me really wants to nudge about doc comments to align with Event Hubs style.   But that's a ton of work and I don't think they're particularly helpful in the context, so I'm going to close my eyes and pretend that I know nothing about this file.